### PR TITLE
Alley ESLint Config

### PR DESCRIPTION
**Pull request by stevenslack**
Alley ESLint config is a set of configurations for configuring ESLint according to Alley's standards. The configuration package provides an easy method to set up configuration depending on a projects setup. This package supports React and Typescript with 4 configurations available.

### CODE REVIEW

File: packages/eslint-config/configs/env.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+module.exports = {
+  env: {
+    browser: true,
+    es2021: true,
```
[the remark this answers is not in the document]
Comment (stevenslack) [reply packages/eslint-config/configs/env.js:4]: esnext?

File: packages/eslint-config/package.json [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    "url": "https://github.com/alleyinteractive/alley-scripts/issues"
+  },
+  "dependencies": {
+    "@babel/core": ">= 7",
+    "@babel/eslint-parser": "^7.21.3",
+    "@babel/preset-env": "^7.20.2",
+    "@typescript-eslint/eslint-plugin": "^5.57.0",
+    "@typescript-eslint/parser": "^5.57.0",
+    "eslint": ">= 8",
+    "eslint-config-airbnb": "^19.0.4",
```
[the remark this answers is not in the document]
Comment (stevenslack) [reply packages/eslint-config/package.json:13]: Yes they are.

File: packages/eslint-config/configs/env.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (kevinfodness) [reply packages/eslint-config/configs/env.js:4]: I'm not seeing an option for `esnext` - it seems like `es2022` is the latest according to the docs: https://eslint.org/docs/latest/use/configure/language-options#specifying-environments

File: packages/block-editor-tools/.eslintrc.json [hunk 3]
```diff
@@ -1,34 +1,7 @@
...
-      "globalReturn": true,
-      "impliedStrict": true,
-      "jsx": true
-    },
-    "project": "./tsconfig.json",
-    "sourceType": "module"
-  },
+  "extends": ["@alleyinteractive/eslint-config/typescript"],
   "rules": {
     "import/no-cycle": "off",
```
[the remark this answers is not in the document]
Comment (stevenslack) [reply packages/block-editor-tools/.eslintrc.json:4]: There are a handful of these rules I have seen developers turning off. I think it makes sense to include those as well, some of which you mentioned above. Just determining the frequency of use and if they deserve to be added is somewhat of a grey area.

File: packages/eslint-config/configs/env.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (stevenslack) [reply packages/eslint-config/configs/env.js]: Got it, I updated the config to `es2022`

**Pull request merged**